src/Controls/src/SourceGen/XmlnsDefinitionSourceGenerator.cs

```diff
@@ -0,0 +1,91 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.Maui.Controls.SourceGen;
+
+using static GeneratorHelpers;
+
+[Generator(LanguageNames.CSharp)]
+public class XmlnsDefinitionSourceGenerator : IIncrementalGenerator
+{
+	public void Initialize(IncrementalGeneratorInitializationContext initContext)
+	{
+		// Read the XmlnsDefinition items data file
+		var xmlnsDefinitionDataProvider = initContext.AdditionalTextsProvider
+			.Where(static file => file.Path.EndsWith(".xmlnsdefinitions", StringComparison.OrdinalIgnoreCase))
+			.Select(static (file, cancellationToken) => ParseXmlnsDefinitionData(file, cancellationToken))
```

Comment:
and GetOptions from there. you will have to declare CompilerVisibleProperties in Common.targets so the sourcegen can see the msbuild properties. see what we do in other SG